graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchDomain.java

```diff
@@ -39,27 +40,27 @@ public SearchDomain(SearchDbService dbService, ViewService viewService) {
         this.viewService = viewService;
     }
 
-    public Optional<Search> getForUser(String id, User user, Predicate<ViewDTO> viewReadPermission) {
+    public Optional<Search> getForUser(String id, SearchUser searchUser) {
         final Optional<Search> search = dbService.get(id);
 
-        search.ifPresent(s -> checkPermission(user, viewReadPermission, s));
+        search.ifPresent(s -> checkPermission(searchUser.username(), searchUser, s));
```

Comment:
checkPermission without String username + searchUser.username() to create the Exception?

Reply:
Good idea! Changed that.